DEVELOPMENT.md

```diff
@@ -28,4 +32,35 @@ convenient formats, for example:
 - To set the `POSTGRES_URL` for all subshells: `eval $(make devenv-export-url`
 
 To permanently configure `POSTGRES_URL` when you change into this directory,
-you may consider using a tool like [direnv](https://direnv.net/).
\ No newline at end of file
+you may consider using a tool like [direnv](https://direnv.net/).
+
+## Testing
+
+Testing in this repository is split into three different kinds of tests, each
+of which are found in their own directory or workspace:
+
+- pgx tests
+- End-to-end SQL tests
+- End-to-end Rust tests
+
+pgx tests are found in the `src` directory, typically in the same source file
+as a function defined in Rust. These tests are solely intended to test the
+behaviour of Postgres functions implemented in Rust.
+
+End-to-end SQL tests are found in the `sql-tests` workspace, in the `testdata`
+directory. The tests are written in SQL and intended to be used to test
+functions which we write in SQL (in the `migration` directory). Each SQL file
```

Comment:
I don't think it's true. For example, we have SQL tests that provide additional coverage for support functions.

```suggestion
functions which we write in SQL (in the `migration` directory) or complex interactions involving both SQL and native functions. Each SQL file
```

Reply:
done